### PR TITLE
fix: undefined ref handling

### PR DESCRIPTION
**Pull request by tushgaurav**
#307 - Fixed ref undefined bug.

### CODE REVIEW

**Comment by arshadkazmi42** [comment]
Thank you for making this change, but the bug is not due to this but due to the wrong heading used in the translation.

Can you update your PR to make that change. This is the diff which will fix this issue

```diff
diff --git a/src/content/learn/choosing-the-state-structure.md b/src/content/learn/choosing-the-state-structure.md
index 71078511..e521a4c3 100644
--- a/src/content/learn/choosing-the-state-structure.md
+++ b/src/content/learn/choosing-the-state-structure.md
@@ -1462,7 +1462,7 @@ button { margin: 10px; }

 <DeepDive>

-#### Improving memory usage {/*improving-memory-usage*/}
+#### मेमोरी यूसेज को अच्छा करना {/*improving-memory-usage*/}

 आदर्श रूप से, आप मेमोरी उपयोग को बेहतर बनाने के लिए हटाए गए आइटम (और उनके चिल्ड्रन!) को "table" ऑब्जेक्ट से हटा देंगे। यह वर्जन ऐसा करता है। यह [Immer का भी उपयोग करता है](/learn/updating-objects-in-state#write-concise-update-logic-with-immer) अपडेट लॉजिक को अधिक संक्षिप्त बनाने के लिए।

@@ -1833,9 +1833,10 @@ button { margin: 10px; }

 <Challenges>

-### अपडेट न होने वाले कौम्पोनॅन्ट को  {/*fix-a-component-thats-not-updating*/}
+#### अपडेट न होने वाले कौम्पोनॅन्ट को  {/*fix-a-component-thats-not-updating*/}

 इस `Clock` कौम्पोनॅन्ट को दो prop मिलते हैं: `color` और `time`। जब आप सिलेक्ट बॉक्स में एक अलग रंग चुनते हैं, `Clock` कौम्पोनॅन्ट अपने पैरेंट कौम्पलग `color` prop प्राप्त करता है। हाललाँकि, किसी कारण से, प्रदर्शित color अपडेट नहहीं होता है। क्ययों? समस्या हल करें।
+
 <Sandpack>

 ```js src/Clock.js active
@@ -1995,7 +1996,7 @@ export default function App() {

 </Solution>

-### टूटी हुई पैककिंग लिस्ट को  {/*fix-a-broken-packing-list*/}
+#### टूटी हुई पैककिंग लिस्ट को  {/*fix-a-broken-packing-list*/}

 इस पैककिंग लिस्ट में एक फुटर है जो दिखता है कि कितने आइटम पैक किए गए हैं और कुल मिलाकर कितने आली बार में काम करने लगता है, लेकिन यह बग्गी है। उदाहरण के लिए, यदि आप किसी आइटम को पैक करते हैं और फिर उसे हटा देते हैं, तो काउंटर ठीक से अपडेट नहहीं होता है। काउंटर को ठीक करें ताकि यह हमेशा सही रहे।

@@ -2279,7 +2280,7 @@ ul, li { margin: 0; padding: 0; }

 </Solution>

-### गायब होने वाले सेलेक्शन को ठ {/*fix-the-disappearing-selection*/}
+#### गायब होने वाले सेलेक्शन को ठ {/*fix-the-disappearing-selection*/}

 State में `letters` की एक लिस्ट है। जबप किसी विशेष अक्षर पर होवर या फ़ोकस करते हैं, तो वह हाइलाइट हो जाता है। अभी हाइलाइट किया गया अक्षर `highlightedLetter` state वेरिएबल में संग्रहित है। आप अलग-अलग अक्षररों को "Star" और "Unstar" कर सकते हैं, जो state में `letters` array को अपडेट करता है।

@@ -2499,7 +2500,7 @@ li { border-radius: 5px; }

 </Solution>

-### मल्टीप्ल सेलेक्शन इम्प्लीमेंट  {/*implement-multiple-selection*/}
+#### मल्टीप्ल सेलेक्शन इम्प्लीमेंट  {/*implement-multiple-selection*/}

 इस उदाहरण में, प्त्येक `Letter` में एक `isSelected` prop और एक `onToggle` handler है जो इसे सेलेक्ट करता है। यह काम करता है, लेकिन state को `selectedId` के रूप में स्टोर किया जाता है (या तो `null` या एक ID), इसलिए किसी भी समय केवल एक ही अक्षर सिलेक्ट किया जा सकता है।
```

**Review by arshadkazmi42 — CHANGES_REQUESTED** [review]
I have added a comment on how to fix this issue.

The case changes are not done in this repository but in English version and are automatically sync in this.

This is broken due to wrong headings used in translation.

File: src/components/MDX/Challenges/Navigation.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ export function Navigation({
     if (containerRef.current) {
       const el = containerRef.current;
       el.scrollLeft =
-        challengesNavRef.current[scrollPos].current?.offsetLeft || 0;
+        challengesNavRef.current[scrollPos]?.current?.offsetLeft || 0;
```
Comment (arshadkazmi42) [inline src/components/MDX/Challenges/Navigation.tsx:68]: can you please revert this change. code changes should not be done here otherwise sync from English repo will fail

**Review by arshadkazmi42 — CHANGES_REQUESTED** [review]
Thank you for making the  change. 

can you revert the change you did in `Navigation.tsx` please

File: src/components/MDX/Challenges/Navigation.tsx (hunk 1 above)
Comment (tushgaurav) [reply src/components/MDX/Challenges/Navigation.tsx]: Oh okay. Done, sorry for the inconvenience.

**Review by arshadkazmi42 — APPROVED** [review]
looks good. thank you for fixing this 💯
